src/transport_ofi.c

```diff
@@ -87,7 +88,20 @@ static char                     myephostname[EPHOSTNAMELEN];
 #endif
 #ifdef ENABLE_THREADS
 shmem_internal_mutex_t          shmem_transport_ofi_lock;
-#endif
+/* Need a syscall to gettid() because glibc doesn't provide a wrapper
+ * (see gettid manpage in the NOTES section): */
+#ifndef __APPLE__
+#ifdef HAVE_SYS_GETTID
+static inline
+TID_TYPE shmem_transport_ofi_gettid(void)
```

Comment:
Is TID_TYPE correct here?  I think the Linux syscall always returns the same type.

Also, why not have a case here that defines ```shmem_transport_ofi_gettid()``` for __APPLE__?  This should simplify things further down and also reduce the total number of ifdefs in this file.

Reply:
I'm not 100% sure on this one.  It seems that nowadays syscall usually returns `long` (but sometimes `int`).  I think it's ok for the return type to be `pid_t`, which is nowadays often 64-bit (sometimes 32, sometimes signed, sometimes unsigned), but I'm ok setting this to `long`, or other ideas.